OSDOCS-14610:Update the z-stream RNs for 4.17.29 (#93320)
openshift/openshift-docs: release_notes/ocp-4-17-release-notes.adoc

* Previously, the Subject Alternative Name (SAN) of the custom certificate that the user added to the `hc.spec.configuration.apiServer.servingCerts.namedCertificates` field conflicted with the hostname set in the `hc.spec.services.servicePublishingStrategy` field for the Kubernetes agent server (KAS). As a result, the KAS certificate was not added to the set of certificates to generate a new payload, causing certificate validation issues for nodes that joined the hosted cluster. With this release, the validation fails earlier so that the user is warned about the issue with the conflicting SANs. (link:https://issues.redhat.com/browse/OCPBUGS-55500[OCPBUGS-55500]). 

* Previously, the boot image updates failed when the Amazon Machine Image (AMI) for a specified region was not found. This issue occurred because AMIs for all regions are not published in the `scos.json` file for the installation program. With this release, update failures are prevented by using the `us-east-1` region by default for any unavailable regions during {aws-first} boot image updates. (link:https://issues.redhat.com/browse/OCPBUGS-55490[OCPBUGS-55490]).

> * Previously, when viewing the list of installed Operators, an Operator appeared twice in the list. This duplication occurred when the currently selected project matched an Operator's default namespace while copied ClusterServiceVersions (CSVs) were disabled in the Operator Lifecycle Manager (OLM). With this release, the Operator appears once. (link:https://issues.redhat.com/browse/OCPBUGS-55415[OCPBUGS-55415]).

Reviewer: `ClusterServiceVersion` is the resource type. A few notes on things like this:
* the resource type goes in backticks: `ClusterServiceVersion`
* we should not use the plural for resource types in docs
* we should not use a resource type (or any other thing that appears in backticks) as a noun in docs. The usual solutions are:
  * use the plain English name: "cluster service version" - this can also be made plural: "cluster service versions"
  * state what the item in backticks is, and use that as the noun. Ex: "list the `MachineSet` resources for the cluster"

This is probably one of the hardest aspects of writing rel notes, because SMEs will not follow these docs conventions and bugs will be from all over the product so they might not be in an area you're familiar with.

For this one, we can just use "cluster service versions (CSVs)".

```suggestion
* Previously, when viewing the list of installed Operators, an Operator appeared twice in the list. This duplication occurred when the currently selected project matched an Operator's default namespace while copied cluster service versions (CSVs) were disabled in the Operator Lifecycle Manager (OLM). With this release, the Operator appears once. (link:https://issues.redhat.com/browse/OCPBUGS-55415[OCPBUGS-55415]).
```

This is good to merge once that's addressed :slightly_smiling_face:

For any {product-title} release, always review the instructions on xref:../updating/updating_a_cluster/updating-cluster-web-console.adoc#updating-cluster-web-console[updating your cluster] properly.
====

// 4.17.29
[id="ocp-4-17-29_{context}"]
=== RHSA-2025:4723 - {product-title} {product-version}.29 bug fix and security update advisory

Issued: 15 May 2025

{product-title} release {product-version}.29 is now available. The list of bug fixes that are included in the update is documented in the link:https://access.redhat.com/errata/RHSA-2025:4723[RHSA-2025:4723] advisory. The RPM packages that are included in the update are provided by the link:https://access.redhat.com/errata/RHBA-2025:4433[RHBA-2025:4725] advisory.

Space precluded documenting all of the container images for this release in the advisory.

You can view the container images in this release by running the following command:

[source,terminal]
----
$ oc adm release info 4.17.29 --pullspecs
----

[id="ocp-4-17-29-bug-fixes_{context}"]
==== Bug fixes

> * Previously, a race condition in the hostname that handled the code caused inconsistencies between the boot and machine hostnames. With this release, the race condition is resolved, which ensures consistent hostnames in the Ignition configuration file during the operating system installation. (link:https://issues.redhat.com/browse/OCPBUGS-55680[OCPBUGS-55680]). 

Reviewer: This text doesn't seem to match the bug linked. It's not really clear when this happens (is "OS installation" referring to some operating system, or OpenShift?) or for what environments - seems it has something to do with the Bare Metal Operator but that isn't mentioned in the writeup :thinking: 

Maybe more like this?
```suggestion
* Previously, a race condition caused inconsistencies between the node and machine hostnames. With this release, the race condition is resolved, which ensures consistent hostnames in the Ignition configuration file during installation. (link:https://issues.redhat.com/browse/OCPBUGS-55680[OCPBUGS-55680]). 
```

That's just based on my read of the bug, should get an SME to check this.

> * Previously, the Subject Alternative Name (SAN) of the custom certificate that the user added to the `hc.spec.configuration.apiServer.servingCerts.namedCertificates` field conflicted with the hostname set in the `hc.spec.services.servicePublishingStrategy` field for the KAS. As a result, the KAS certificate was not added to the set of certificates to generate a new payload, causing new nodes that joined the HostedCluster to have validation problems with the certificates. With this release, the validation will fail earlier so that the user is warned about the issue with the conflicting SANs. (link:https://issues.redhat.com/browse/OCPBUGS-55500[OCPBUGS-55500]). 

Reviewer: * What's KAS? Define on first appearance
* where is the `hc.spec.configuration.apiServer.servingCerts.namedCertificates` field?

```suggestion
* Previously, the Subject Alternative Name (SAN) of the custom certificate that the user added to the `hc.spec.configuration.apiServer.servingCerts.namedCertificates` field conflicted with the hostname set in the `hc.spec.services.servicePublishingStrategy` field for the KAS. As a result, the KAS certificate was not added to the set of certificates to generate a new payload, causing certificate validation issues for nodes that joined the hosted cluster. With this release, the validation fails earlier so that the user is warned about the issue with the conflicting SANs. (link:https://issues.redhat.com/browse/OCPBUGS-55500[OCPBUGS-55500]). 
```

> * Previously, the boot image updates failed when the AMI for a specified region was not found because, in OKD, Amazon Machine Images (AMIs) for all regions are not published in the `scos.json` file for the installer. With this release, update failures do not occur because `us-east-1` is the default region that is used for the unavailable regions during Amazon Web Services (AWS) boot image updates. (link:https://issues.redhat.com/browse/OCPBUGS-55490[OCPBUGS-55490]).

Reviewer: ```suggestion
* Previously, the boot image updates failed when the Amazon Machine Image (AMI) for a specified region was not found. This issue occurred because, in OKD, AMIs for all regions are not published in the `scos.json` file for the installation program. With this release, update failures are prevented by using the `us-east-1` region by default for any unavailable regions during {aws-first} boot image updates. (link:https://issues.redhat.com/browse/OCPBUGS-55490[OCPBUGS-55490]).
```

OKD is the upstream version of OpenShift Container Platform. Typically, I would expect to see `{product-title}` rather than OKD, which would render as OCP or OKD as appropriate per each doc set. Please check on this use of "OKD" here and see if it should be `{product-title}` so that the OCP RNs say "OpenShift Container Platform" instead. Really, since these are the OCP rel notes, it might make sense to omit `, in OKD,` entirely:
```suggestion
* Previously, the boot image updates failed when the Amazon Machine Image (AMI) for a specified region was not found. This issue occurred because AMIs for all regions are not published in the `scos.json` file for the installation program. With this release, update failures are prevented by using the `us-east-1` region by default for any unavailable regions during {aws-first} boot image updates. (link:https://issues.redhat.com/browse/OCPBUGS-55490[OCPBUGS-55490]).
```

> * Previously, when viewing the list of installed operators and the currently selected project matches an operator's default namespace while copied CSVs are disabled in OLM, an operator appeared twice in the list. With this release, the operator appears once. (link:https://issues.redhat.com/browse/OCPBUGS-55415[OCPBUGS-55415]).

Reviewer: * Operator is always capitalized
* Can that first sentence be broken into 2-3? it's quite long and complex.
* Define OLM and CSV?
```suggestion
* Previously, when viewing the list of installed Operators and the currently selected project matches an Operator's default namespace while copied CSVs are disabled in OLM, an Operator appeared twice in the list. With this release, the Operator appears once. (link:https://issues.redhat.com/browse/OCPBUGS-55415[OCPBUGS-55415]).
```
